lib/src/model/common/uci.dart

```diff
@@ -37,7 +37,7 @@ class UciCharPair with _$UciCharPair {
             String.fromCharCode(35 + f),
             String.fromCharCode(
               p != null
-                  ? 35 + 64 + 8 * _promotionRoles.indexOf(p) + squareFile(t)
+                  ? 35 + 64 + 8 * _promotionRoles.indexOf(p) + t.file.value
```

Comment:
Since the `File` extension type implements `int`, unwrapping `.value` is not necessary.